aws/src/main/java/org/apache/iceberg/aws/AwsClientFactories.java

```diff
@@ -50,7 +51,16 @@ public static AwsClientFactory defaultFactory() {
   public static AwsClientFactory from(Map<String, String> properties) {
     String factoryImpl = PropertyUtil.propertyAsString(
         properties, AwsProperties.CLIENT_FACTORY, DefaultAwsClientFactory.class.getName());
-    return loadClientFactory(factoryImpl, properties);
+    AwsClientFactory awsClientFactory = loadClientFactory(factoryImpl, properties);
+    if (PropertyUtil.propertyAsBoolean(
+        properties,
+        AwsProperties.GLUE_LAKEFORMATION_ENABLED,
+        AwsProperties.GLUE_LAKEFORMATION_ENABLED_DEFAULT)) {
+      Preconditions.checkArgument((awsClientFactory instanceof LakeFormationAwsClientFactory),
```

Comment:
nit: do we need the () around `awsClientFactory instanceof LakeFormationAwsClientFactory`?